apps/browser/src/autofill/services/inline-menu-field-qualification.service.ts

```diff
@@ -250,7 +290,13 @@ export class InlineMenuFieldQualificationService
    * @param field - The field to validate
    */
   private isNewPasswordField = (field: AutofillField): boolean => {
-    if (field.autoCompleteType === "current-password") {
+    if (
+      this.fieldContainsAutocompleteValues(
+        field.autoCompleteType,
+        this.currentPasswordAutocompleteValues,
+      ) ||
+      !this.keywordsFoundInFieldData(field, [...this.accountCreationFieldKeywords])
+    ) {
```

Comment:
I would drop the negative `keywordsFoundInFieldData` check here if you can

```suggestion
    if (
      this.fieldContainsAutocompleteValues(
        field.autoCompleteType,
        this.currentPasswordAutocompleteValues,
      )
    ) {
```

```
this.keywordsFoundInFieldData(field, [...this.accountCreationFieldKeywords])
```
Is a relatively authoritative way to identify an account creation field.

```
!this.keywordsFoundInFieldData(field, [...this.accountCreationFieldKeywords])
```
is not - it is inferred it from the lack of appearing to be an account creation field, which is much less authoritative (all it says is that it does not _definitively_ appear to be an account creation field; that doesn't mean it still isn't)

Reply:
A more authoritative equivalent would be something like `!this.keywordsFoundInFieldData(field, [...this.accountLoginFieldKeywords])` with  existing account terms (e.g. "login" as seen on the CNN website)

Reply:
The consequence of more easily discarding password inputs as account creation inputs would be potentially breaking autofill on password inputs that currently fill correctly but are simply ambiguous from a programmatic evaluation standpoint.